File: src/api/digitalBuddyApi.ts
import type { BuddyLanguage } from "../components/DigitalBuddy/buddyMessages";
import { httpClient } from "./httpClient";

export type DigitalBuddyAnswer = {
  answer: string;
  source?: string;
  section?: string;
  document_code?: string;
  sentiment?: string;
  language?: "ru" | "kk";
  mode?: "llm" | "rag" | "fallback" | "no_context" | "chat";
};

export type DigitalBuddyStatus = {
  llm_enabled: boolean;
  llm_available: boolean;
  llm_provider: string;
  llm_model: string;
  model_ready: boolean;
  installed_models: string[];
  embedding_model?: string;
  embedding_model_ready?: boolean;
  chunks_count?: number;
  indexed_documents?: string[];
  chroma_ready?: boolean;
  chroma_count?: number;
  last_indexed_at?: string;
};

export const digitalBuddyApi = {
  async getStatus(): Promise<DigitalBuddyStatus> {
    const response = await httpClient.get<DigitalBuddyStatus>(
      "/digital-buddy/status"
    );

    return response.data;
  },

  async askQuestion(
    question: string,
    language?: BuddyLanguage,
  ): Promise<DigitalBuddyAnswer> {
    const response = await httpClient.post<DigitalBuddyAnswer>(
      "/digital-buddy/ask",
      {
        employee_id: 1,
        question,
        language,
      }
    );

    return response.data;
  },
};

File: mobile/lib/api/digital_buddy_api.dart
import '../core/network/api_client.dart';
import '../models/buddy.dart';

/// Порт src/api/digitalBuddyApi.ts.
class DigitalBuddyApi {
  const DigitalBuddyApi();

  Future<BuddyStatus> getStatus() async {
    final response = await apiClient.get('/digital-buddy/status');
    return BuddyStatus.fromJson(response.data as Map<String, dynamic>);
  }

  Future<BuddyAnswer> askQuestion(
    int employeeId,
    String question, {
    String? language,
  }) async {
    final response = await apiClient.post(
      '/digital-buddy/ask',
      data: {
        'employee_id': employeeId,
        'question': question,
        'language': ?language,
      },
    );
    return BuddyAnswer.fromJson(response.data as Map<String, dynamic>);
  }
}

const digitalBuddyApi = DigitalBuddyApi();
